### PR TITLE
adds design tokens

### PR DESCRIPTION
**Pull request by sturobson**
as discussed 'way back when' here's the first pass of the workflow for creating and integrating design tokens for springer nature products using the front-end toolkits.

This PR does a few things:

- [x] creates design tokens for colours, typography, spacing, breakpoints
- [x] generates a `index.scss` file that imports the compiled Sass files so it's a single import for the toolkits.
- [x] a decision on how best this becomes a package
- [x] is included as part of the front-end toolkits
- [x] has alias tokens for actual use. (phase 2 work)

### CODE REVIEW

**Comment by sturobson** [comment]
Screenshot of the branch conflict message with the package-lock.json

more and more I feel that the introduction of a package-lock.json file was a mistake and only implemented to 'keep up' with Yarn.

**Comment by Heydon** [comment]
> more and more I feel that the introduction of a package-lock.json file was a mistake and only implemented to 'keep up' with Yarn.

+10000000000000000000000

**Comment by aliuk2012** [comment]
🤔  I see you only adding `style-dictionary` package. Could that maybe a separate commit/PR so that it's less likely there will be a merge conflicts with the lock file? I know it will not be used at first but it at least lays the ground work?

File: context/brand-context/default/scss/00-tokens/_background.variables.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+// Do not edit directly
+// Generated on Wed, 27 Apr 2022 14:51:22 GMT
+
+$background-page: #f8f8f8;
+$background-container: color.white};
```
Comment (alexkilgour) [inline context/brand-context/default/scss/00-tokens/_background.variables.scss:6]: looks like a stray `}`

File: context/brand-context/nature/scss/00-tokens/_illustration.variables.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+// Do not edit directly
+// Generated on Wed, 27 Apr 2022 14:51:22 GMT
+
+$illustration-background-001: #29303C;
+$illustration-background-002: {nature-color.gunmetal.500;
```
Comment (alexkilgour) [inline context/brand-context/nature/scss/00-tokens/_illustration.variables.scss:6]: another stray `{`, not sure about the rest?

File: context/brand-context/default/scss/00-tokens/_background.variables.scss (hunk 1 above)
Comment (sturobson) [reply context/brand-context/default/scss/00-tokens/_background.variables.scss:6]: I've got this fixed locally now - will push a new commit in a bit

File: context/brand-context/nature/scss/00-tokens/_illustration.variables.scss (hunk 2 above)
Comment (sturobson) [reply context/brand-context/nature/scss/00-tokens/_illustration.variables.scss:6]: I've got this fixed locally now - will push a new commit in a bit

Comment (sturobson) [reply context/brand-context/nature/scss/00-tokens/_illustration.variables.scss]: fixed in [this commit](https://github.com/springernature/frontend-toolkits/pull/735/commits/c7cdb34c3e9db545055ae6c6577c19a57f563732) and [this other commit](https://github.com/springernature/frontend-toolkits/pull/735/commits/49a0d698182ac67bbec4b892f4cc8f21883a04da)

File: context/brand-context/default/scss/00-tokens/_background.variables.scss (hunk 1 above)
Comment (sturobson) [reply context/brand-context/default/scss/00-tokens/_background.variables.scss:6]: fixed in [this commit](https://github.com/springernature/frontend-toolkits/pull/735/commits/c7cdb34c3e9db545055ae6c6577c19a57f563732) and [this other commit](https://github.com/springernature/frontend-toolkits/pull/735/commits/49a0d698182ac67bbec4b892f4cc8f21883a04da)

File: context/brand-context/default/scss/00-tokens/_background.variables.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+
+// Do not edit directly
+// Generated on Wed, 27 Apr 2022 15:18:06 GMT
+
+$token-background-page: #f8f8f8;
```
[the remark this answers is not in the document]
Comment (sturobson) [reply context/brand-context/default/scss/00-tokens/_background.variables.scss:5]: Yes, Style-Dictionary allows for a prefix to be added and stipulated in the config.

I've added [`token`]([f9dbbfc](https://github.com/springernature/frontend-toolkits/pull/735/commits/f9dbbfcec02c8bfc751a8e10bd0a843f269a4b93)) so we get [`$token-` as a prefix for the Sass variables]([e3eb298](https://github.com/springernature/frontend-toolkits/pull/735/commits/e3eb29849e57806cf503c1175f085c6f787488b2)) and committed.

It makes the variables kinda long -- but I think we can go with this for now and when moving to the wholesale versioning integrating the tokens into components the length of variable name is a little negated.

**Review by alexkilgour — APPROVED** [review]
looking great!

**Pull request merged**